src/ExpressionEvaluator/Core/Source/ExpressionCompiler/MetadataUtilities.cs

```diff
@@ -113,6 +113,71 @@ internal static ImmutableArray<MetadataReference> MakeAssemblyReferences(this Im
             return referencesBuilder.ToImmutableAndFree();
         }
 
+        private static void RemoveStrongNamedDuplicates(ArrayBuilder<MetadataReference> references, ArrayBuilder<AssemblyIdentity> identities)
+        {
+            Debug.Assert(references.Count == identities.Count);
+            Debug.Assert(identities.All(i => i != null));
+
+            // Null out any identities that are duplicates. Note, this is O(n^2).
+            int n = references.Count;
+            for (int i = 0; i < n; i++)
+            {
+                if (!IsNonNullAndStrongNamed(identities[i]))
+                {
+                    continue;
+                }
+                var bestIndex = i;
+                for (int j = i + 1; j < n; j++)
+                {
+                    if (!IsNonNullAndStrongNamed(identities[j]))
+                    {
+                        continue;
+                    }
+                    var compareResult = AssemblyIdentityComparer.Default.Compare(identities[bestIndex], identities[j]);
```

Comment:
Should the AssemblyIdentityComparer be a parameter?  I think the compilation (options?) has one.